moveit_common/cmake/moveit_package.cmake

```diff
@@ -46,6 +46,16 @@ macro(moveit_package)
     add_compile_options(-Wall -Wextra
       -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wcast-qual
       -Wno-unused-parameter -Wno-unused-function)
+
+    # Use lld linker if it is installed
+    execute_process(COMMAND ${CMAKE_CXX_COMPILER} -fuse-ld=lld -Wl,--version OUTPUT_VARIABLE lld_test ERROR_QUIET)
```

Comment:
This feels like too much automation, I would simply set the variables in the job.
Otherwise users with both linkers installed might wonder what changed or if they have a linker set on the command line it will get overwritten.

Reply:
I wasn't sure if anyone would care what linker ran.  And this way I give everyone (who has lld installed) faster builds without them doing anything.

I didn't know there was alternatives to ld until this week.  In my testing lld seems to perform slightly better than gold and both of them are much much faster than ld.

Reply:
Well, I guess most people don't care but we can't guarantee that it won't break moveit for some. If you really want to change the behavior for everyone, I would ask for more tests then this. Given that uses can change it easily in thair environment, I would stick with the default.

Reply:
I opened another PR with the change only being made to the CI config.  Let me know if you think that is a better approach.